docker-storage-setup.sh

```diff
@@ -379,27 +360,25 @@ create_data_lv() {
 
   check_min_data_size_condition
 
-  # TODO: Error handling when DATA_SIZE > available space.
-  if [[ $DATA_SIZE == *%* ]]; then
-    lvcreate -y -l $DATA_SIZE -n $DATA_LV_NAME $VG
-  else
-    lvcreate -y -L $DATA_SIZE -n $DATA_LV_NAME $VG
-  fi
-}
-
-create_lvm_thin_pool () {
-  if [ -z "$DEVS_ABS" ] && [ -z "$VG_EXISTS" ]; then
-    Fatal "Specified volume group $VG does not exist, and no devices were specified"
+  # Calculate size of metadata lv. Reserve 0.1% of the free space in the VG
+  # for docker metadata.
+  VG_SIZE=$(vgs --noheadings --nosuffix --units s -o vg_size $VG)
+  META_SIZE=$(( $VG_SIZE / 1000 + 1 ))
+  if [ -z "$META_SIZE" ];then
+    Fatal "Failed to calculate size of metadata volume"
```

Comment:
> We had a long conversation in the past about how big a metadata lvm should be and we did not have a good answer.

Even more of a reason to let the tool decide.

> Even if we have to do it, it should be done in a separate PR and this PR should not be blocked because of that.

I can accept that. if we leave it like this then we should probably add the 'if statement' back